Src/Newtonsoft.Json/Utilities/TypeExtensions.cs

```diff
@@ -135,7 +135,7 @@ public static MemberTypes MemberType(this MemberInfo memberInfo)
 
         public static bool ContainsGenericParameters(this Type type)
         {
-#if !(DOTNET || PORTABLE)
+#if HAVE_FULL_REFLECTION
```

Comment:
`HAVE_FULL_REFLECTION` means that the `GetTypeInfo` indirection is not necessary. `HAVE_REFLECTION` is used in `DynamicUtils` where it basically checks the availability of `System.Reflection.Binder`. I'll change it to `HAVE_REFLECTION_BINDER`.